l1-contracts-foundry/script/DeployHyperchain.s.sol

```diff
@@ -0,0 +1,261 @@
+        facetCuts[0] = Diamond.FacetCut({
+            facet: config.addresses.adminFacet,
+            action: Diamond.Action.Add,
+            isFreezable: false,
+            selectors: Utils.getAllSelectors(config.addresses.adminFacet.code)
+        });
+        facetCuts[1] = Diamond.FacetCut({
+            facet: config.addresses.gettersFacet,
+            action: Diamond.Action.Add,
+            isFreezable: false,
+            selectors: Utils.getAllSelectors(config.addresses.gettersFacet.code)
+        });
+        facetCuts[2] = Diamond.FacetCut({
+            facet: config.addresses.mailboxFacet,
+            action: Diamond.Action.Add,
+            isFreezable: true,
+            selectors: Utils.getAllSelectors(config.addresses.mailboxFacet.code)
+        });
+        facetCuts[3] = Diamond.FacetCut({
+            facet: config.addresses.executorFacet,
+            action: Diamond.Action.Add,
+            isFreezable: true,
+            selectors: Utils.getAllSelectors(config.addresses.executorFacet.code)
+        });
+
+        VerifierParams memory verifierParams = VerifierParams({
+            recursionNodeLevelVkHash: config.contracts.recursionNodeLevelVkHash,
+            recursionLeafLevelVkHash: config.contracts.recursionLeafLevelVkHash,
+            recursionCircuitsSetVksHash: config.contracts.recursionCircuitsSetVksHash
+        });
+
+        FeeParams memory feeParams = FeeParams({
+            pubdataPricingMode: PubdataPricingMode.Rollup,
+            batchOverheadL1Gas: uint32(config.contracts.diamondInitBatchOverheadL1Gas),
+            maxPubdataPerBatch: uint32(config.contracts.diamondInitMaxPubdataPerBatch),
+            maxL2GasPerBatch: uint32(config.contracts.diamondInitMaxL2GasPerBatch),
+            priorityTxMaxPubdata: uint32(config.contracts.diamondInitPriorityTxMaxPubdata),
+            minimalL2GasPrice: uint64(config.contracts.diamondInitMinimalL2GasPrice)
+        });
+
+        DiamondInitializeDataNewChain memory initializeData = DiamondInitializeDataNewChain({
+            verifier: IVerifier(config.addresses.verifier),
+            verifierParams: verifierParams,
+            l2BootloaderBytecodeHash: bytes32(Utils.getBatchBootloaderBytecodeHash()),
+            l2DefaultAccountBytecodeHash: bytes32(Utils.readSystemContractsBytecode("DefaultAccount")),
+            priorityTxMaxGasLimit: config.contracts.priorityTxMaxGasLimit,
+            feeParams: feeParams,
+            blobVersionedHashRetriever: config.addresses.blobVersionedHashRetriever
+        });
+
+        Diamond.DiamondCutData memory initData = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: config.addresses.diamondInit,
+            initCalldata: abi.encode(initializeData)
+        });
+
+        IBridgehub bridgehub = IBridgehub(config.addresses.bridgehub);
+
+        vm.broadcast();
+        vm.recordLogs();
+        bridgehub.createNewChain({
+            _chainId: config.eraChainId,
```

Comment:
I think it suppose to be not eraChainId, but hyperchainChainId

Reply:
Where do you suggest we should pull this config from? Right now we're getting all the configs from the output of DeployL1 script. Everything else is hardcoded to avoid having another config file for DeployHyperchain.